src/pages/iou/MoneyRequestDescriptionPage.js

```diff
@@ -45,10 +47,14 @@ class MoneyRequestDescriptionPage extends Component {
         this.reportID = lodashGet(props.route, 'params.reportID', '');
     }
 
-    // eslint-disable-next-line rulesdir/prefer-early-return
     componentDidMount() {
-        if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0) {
-            Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID));
+        const moneyRequestId = `${this.iouType}${this.reportID}`;
+        const shouldReset = this.props.iou.id !== moneyRequestId;
+        if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0 || shouldReset) {
+            if (shouldReset) {
+                IOU.resetMoneyRequestInfo(moneyRequestId);
+            }
+            Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), shouldReset);
```

Comment:
Why we are enforcing this fallback only if `shouldReset` is true?

Reply:
We are resetting the money request data and want to force the user to be navigated to the first page of money request when the id changes.

Reply:
Clear thanks!